Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/OnUpdate.hs

```diff
@@ -333,6 +335,8 @@ onUpdate ValidatedRideCompletedReq {..} = do
   _ <- QRide.updateMultiple updRide.id updRide
   _ <- QFareBreakup.createMany breakups
   void $ QPFS.updateStatus booking.riderId DPFS.PENDING_RATING {rideId = ride.id}
+  when (person.cancellationDues > 0) $ do
+    QP.updateDisputeChancesUsed person.id 0 >> QP.updateCancellationDues person.id 0
```

Comment:
Why is this added here?

Reply:
when due amount is paid we will revive his chances and dues will be 0 in endRide

Reply:
What ensures due amount is paid here?

Reply:
After the ride customer has to pay the amount and this amount is added in the ride fare